### PR TITLE
SDIT-1367 Add delete incident mapping endpoints

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt [hunk 1]
```diff
@@ -542,13 +542,203 @@ class IncidentMappingResourceIntTest : IntegrationTestBase() {
...
+        .exchange()
+        .expectStatus().isOk
+
+      // delete mapping
+      webTestClient.delete().uri("/mapping/incidents/incident-id/$INCIDENT_ID")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCIDENTS")))
+        .exchange()
+        .expectStatus().isNoContent
+
+      // no longer present by nonAssociation id
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt:606]: evidence of copy and paste !

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt [hunk 2]
```diff
@@ -134,21 +134,21 @@ class IncidentMappingResourceIntTest : IntegrationTestBase() {
         .returnResult().responseBody
 
       with(responseBody!!) {
-        assertThat(userMessage).contains("Conflict: Incident mapping already exists.\nExisting mapping: IncidentMappingDto(nomisIncidentId=1234, incidentId=4444, label=2022-01-01, mappingType=NOMIS_CREATED")
-        assertThat(userMessage).contains("Duplicate mapping: IncidentMappingDto(nomisIncidentId=21, incidentId=4444, label=2022-01-01, mappingType=NOMIS_CREATED, whenCreated=null)")
+        assertThat(userMessage).contains("Conflict: Incident mapping already exists.\nExisting mapping: IncidentMappingDto(nomisIncidentId=1234, incidentId=4321, label=2022-01-01, mappingType=NOMIS_CREATED")
+        assertThat(userMessage).contains("Duplicate mapping: IncidentMappingDto(nomisIncidentId=21, incidentId=4321, label=2022-01-01, mappingType=NOMIS_CREATED, whenCreated=null)")
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt:138]: very finickity (and other syncs have this too) but i suppose these '4321's should be $INCIDENT_ID

**Review by steverendell — APPROVED** [review]
lgtm

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt (hunk 2 above)
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt:138]: yes - I did a full pull out of these to make it more identifiable - will update in next PR

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt (hunk 1 above)
Comment (karenmillermoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/IncidentMappingResourceIntTest.kt:606]: will fix

**Pull request merged**